### PR TITLE
SAAS-3063: Added Timestream Databases Encrypted Plugin

### PR DESCRIPTION
**Pull request by sadeed12345**

### CODE REVIEW

File: plugins/aws/timestreamwrite/timestreamDatabaseEncrypted.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+                    if (!describeKey || describeKey.err || !describeKey.data || !describeKey.data.KeyMetadata) {
+                        helpers.addResult(results, 3,
+                            `Unable to query KMS key: ${helpers.addError(describeKey)}`,
+                            region, kmsKeyId);
+                        continue;
+                    }
+
+                    currentEncryptionLevel = helpers.getEncryptionLevel(describeKey.data.KeyMetadata, helpers.ENCRYPTION_LEVELS);
+                } else {
+
```
Comment (sadeed12345) [inline plugins/aws/timestreamwrite/timestreamDatabaseEncrypted.js:81]:
```suggestion
```

File: plugins/aws/timestreamwrite/timestreamDatabaseEncrypted.spec.js [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+        },
+        {
+            "Arn": "arn:aws:timestream:us-east-1:000011112222:database/sampleDB",
+            "DatabaseName": "sampleDB",
+            "TableCount": 0,
+            "KmsKeyId": "arn:aws:kms:us-east-1:000011112222:key/c4750c1a-72e5-4d16-bc72-0e7b559e0250",
+            "CreationTime": "2021-11-12T02:59:33.357000-08:00",
+            "LastUpdatedTime": "2021-11-12T02:59:33.357000-08:00"
+        }
+
```
Comment (sadeed12345) [inline plugins/aws/timestreamwrite/timestreamDatabaseEncrypted.spec.js:21]:
```suggestion
```

File: plugins/aws/timestreamwrite/timestreamDatabaseEncrypted.spec.js [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+        {
+            "Arn": "arn:aws:timestream:us-east-1:000011112222:database/sampleDB",
+            "DatabaseName": "sampleDB",
+            "TableCount": 0,
+            "KmsKeyId": "arn:aws:kms:us-east-1:000011112222:key/c4750c1a-72e5-4d16-bc72-0e7b559e0250",
+            "CreationTime": "2021-11-12T02:59:33.357000-08:00",
+            "LastUpdatedTime": "2021-11-12T02:59:33.357000-08:00"
+        }
+    
+
```
Comment (sadeed12345) [inline plugins/aws/timestreamwrite/timestreamDatabaseEncrypted.spec.js:21]:
```suggestion
```

File: plugins/aws/timestreamwrite/timestreamDatabaseEncrypted.spec.js [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+            const cache = createCache([listDatabases[0]], listKeys, describeKey[1]);
+            timestreamDatabaseEncrypted.run(cache, { timestream_databases_encryption:'awscmk' }, (err, results) => {
+                expect(results.length).to.equal(1);
+                expect(results[0].status).to.equal(2);
+                expect(results[0].message).to.include('Timestream Databasesis encrypted with awskms');
+                done();
+            });
+        });
+
+
```
Comment (sadeed12345) [inline plugins/aws/timestreamwrite/timestreamDatabaseEncrypted.spec.js:123]:
```suggestion
```

File: plugins/aws/timestreamwrite/timestreamDatabaseEncrypted.js [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+                if (database.KmsKeyId) {
+                    var kmsKeyId = database.KmsKeyId.split('/')[1] ? database.KmsKeyId.split('/')[1] : database.KmsKeyId;
+
+                    var describeKey = helpers.addSource(cache, source,
+                        ['kms', 'describeKey', region, kmsKeyId]);  
+
+                    if (!describeKey || describeKey.err || !describeKey.data || !describeKey.data.KeyMetadata) {
+                        helpers.addResult(results, 3,
+                            `Unable to query KMS key: ${helpers.addError(describeKey)}`,
+                            region, kmsKeyId);
```
Comment (dipsubha06) [inline plugins/aws/timestreamwrite/timestreamDatabaseEncrypted.js:74]:
```suggestion
                            region, database.KmsKeyId);
```

File: plugins/aws/timestreamwrite/timestreamDatabaseEncrypted.spec.js [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+                expect(results.length).to.equal(1);
+                expect(results[0].status).to.equal(0);
+                expect(results[0].message).to.include('Timestream database is encrypted with awscmk');
+                done();
+            });
+        });
+
+
+        it('should FAIL if Timestream database is not encrypted with desired encyption level', function (done) {
+            const cache = createCache([listDatabases[0]], listKeys, describeKey[1]);
```
Comment (dipsubha06) [inline plugins/aws/timestreamwrite/timestreamDatabaseEncrypted.spec.js:114]: Here the keys are different in listDatabases[0] and describeKey[1]. Please check.

File: plugins/aws/timestreamwrite/timestreamDatabaseEncrypted.spec.js [hunk 7]
```diff
@@ -0,0 +1,152 @@
+var expect = require('chai').expect;
+var timestreamDatabaseEncrypted = require('./timestreamDatabaseEncrypted');
+
+const listDatabases = [
+        {
+            "Arn": "arn:aws:timestream:us-east-1:000011112222:database/akhtar-db1",
+            "DatabaseName": "akhtar-db1",
+            "TableCount": 0,
+            "KmsKeyId": "arn:aws:kms:us-east-1:000011112222:key/b6d7913b-5604-47c5-8291-8270a1abab58",
```
Comment (dipsubha06) [inline plugins/aws/timestreamwrite/timestreamDatabaseEncrypted.spec.js:9]: The key arn is different than all other arns.

**Review by dipsubha06 — CHANGES_REQUESTED** [review]

**Review by dipsubha06 — APPROVED** [review]

**Pull request merged**